tensorflow/python/autograph/operators/py_builtins_test.py

```diff
@@ -137,6 +139,30 @@ def test_range_tensor_empty_range(self):
       r = py_builtins.range_(5, constant_op.constant(2))
       self.assertAllEqual(self.evaluate(r), [])
 
+  def test_enumerate(self):
+    self.assertListEqual(list(py_builtins.enumerate_([3,2,1])), [(0, 3), (1, 2), (2, 1)])
+    self.assertListEqual(list(py_builtins.enumerate_([3,2,1], 5)), [(5, 3), (6, 2), (7, 1)])
+    self.assertListEqual(list(py_builtins.enumerate_([-8], -3)), [(-3, -8)])
+
+  @test_util.run_all_in_graph_and_eager_modes
+  def test_enumerate_dataset(self):
+    components = (["a", "b"], [1, 2], [37.0, 38])
+    start = constant_op.constant(20, dtype=dtypes.int64)
+
+    dataset = py_builtins.enumerate_(dataset_ops.Dataset.from_tensor_slices(
+        components), start)
+
+    self.assertEqual(dtypes.int64,
+                     dataset_ops.get_legacy_output_types(dataset)[0])
+    dataset_output_shapes = dataset_ops.get_legacy_output_shapes(dataset)
+    self.assertEqual((), dataset_output_shapes[0])
+    self.assertEqual([tensor_shape.TensorShape([])] * 3,
+                     [shape for shape in dataset_output_shapes[1]])
+
+    self.assertDatasetProduces(dataset, [(20, (b"a", 1, 37.0)),
+                                         (21, (b"b", 2, 38.0))])
```

Comment:
I don't think assertDatasetProduces is available here - is the test passing?

Anyway, it it doesn't pass, it should be sufficient to do something simpler, like this:

```
input = dataset_ops.DatasetV2.from_tensor_slices(["a", "b"])
start = constant_op.constant(20, dtype=dtypes.int64)

dataset = py_builtins.enumerate_(input, start)

first_element = next(iter(dataset))
self.assertAllEqual(self.evaluate(first_element), (20, "a"))
```

Reply:
Ah, my bad. I'm so sorry. Yes that the one you suggest is sufficient. I've updated the test case with a working one.